pipelines/library/src/ParseVersion.groovy

```diff
@@ -0,0 +1,52 @@
+import common.VersionInfo
+import groovy.json.JsonOutput
+
+import java.util.regex.Matcher
+
+class ParseVersion {
+
+    static void main(String[] args) {
+        OptionAccessor parsedArgs = parseArgs(args)
+        VersionInfo version
+        if (parsedArgs.s) {
+            version = parseFromStdIn(args)
+        } else {
+            version = new VersionInfo().parse(args[args.length - 2], args[args.length - 1])
+        }
+
+        printVersion(parsedArgs, version)
+    }
+
+    private static void printVersion(OptionAccessor parsedArgs, VersionInfo version) {
+        if (parsedArgs.f) {
```

Comment:
`f`?

Reply:
the sort of interesting way groovys CliBuilder works. If the user gave a `-f` argument then `parsedArgs.f` will exist